code/go/0chain.net/smartcontract/dbs/event/eventdb.go

```diff
@@ -68,6 +69,7 @@ type EventDb struct {
 	dbConfig      config.DbAccess   // depends on the sharder, change on restart
 	settings      config.DbSettings // the same across all sharders, needs to mirror blockchain
 	eventsChannel chan BlockEvents
+	kafka         *queueProvider.KafkaProvider
```

Comment:
we need to initialize kafka on some deployments, on others we need to use no-op stub